_integration-tests/utils/testcontainers.go

```diff
@@ -75,6 +88,39 @@ func StartKafkaTestContainer(t *testing.T) (*kafka.KafkaContainer, string) {
 	return container, addr
 }
 
+// StartRedisTestContainer starts a new Redis test container and returns the connection string.
+func StartRedisTestContainer(t *testing.T) (*redis.RedisContainer, string) {
+	ctx := context.Background()
+	exposedPort := "6379/tcp"
+	waitReadyCmd := []string{
+		"redis-cli",
+		"ping",
+	}
+
+	container, err := redis.Run(ctx,
+		"redis:7",
+		testcontainers.WithLogger(testcontainers.TestLogger(t)),
+		WithTestLogConsumer(t),
+		testcontainers.WithWaitStrategy(
+			wait.ForAll(
+				wait.ForLog("* Ready to accept connections"),
+				wait.ForExposedPort(),
+				wait.ForListeningPort(nat.Port(exposedPort)),
+				wait.ForExec(waitReadyCmd),
+			),
+		),
+	)
+	AssertTestContainersError(t, err)
+
+	connStr, err := container.ConnectionString(ctx)
+	require.NoError(t, err)
+
+	redisURL, err := url.Parse(connStr)
+	require.NoError(t, err)
```

Comment:
So I guess if these fail; the test will just bail out right away and not clean up the container...